### ISSUE TITLE
Uncaught SecurityError: Blocked a frame with origin X from accessing a frame with origin Y

### ISSUE THREAD

**Issue by avzabr** [KartikTalwar/gmail.js#295]
I writing the plugin to chrome similar to [gmail-chrome-extension-boilerplate](https://github.com/KartikTalwar/gmail-chrome-extension-boilerplate) and get the following error:

```
gmail.js:10 Uncaught SecurityError: Blocked a frame with origin "https://mail.google.com" from accessing a frame with origin "https://contacts.google.com". Protocols, domains, and ports must match.Gmail_ 
@ gmail.js:10(anonymous function) 
@ gmail.js:10main 
```

after the steps:
- Install chrome extension similar to [gmail-chrome-extension-boilerplate](https://github.com/KartikTalwar/gmail-chrome-extension-boilerplate)
- Navigate to gmail
-  Click contacts 
  ![contacts_1](https://cloud.githubusercontent.com/assets/17640566/17735754/bb960de0-648c-11e6-9cc4-fb74ae132051.png)
- From contacts click on any email to open compose screen in a frame
  ![contacts_2](https://cloud.githubusercontent.com/assets/17640566/17735772/d0564826-648c-11e6-9f1c-809978d3f11c.png)
- Error is in console and script is not loaded
  ![contacts_3](https://cloud.githubusercontent.com/assets/17640566/17735790/e7e577aa-648c-11e6-97ac-e9f8d5ed74c6.png)
  ![contacts_4](https://cloud.githubusercontent.com/assets/17640566/17735791/e7e90a64-648c-11e6-9365-8ecb6cd620fe.png)

Temporary workaround (at least to load script and do not break main.js) is to surround the  [block of code](https://github.com/KartikTalwar/gmail.js/blob/master/src/gmail.js#L34-L35) with try/catch:

```
try {
api.tracker.globals   = typeof GLOBALS !== 'undefined' ? GLOBALS : ( window.opener != null && typeof window.opener.GLOBALS !== 'undefined' ? window.opener.GLOBALS : [] );
api.tracker.view_data = typeof VIEW_DATA !== 'undefined' ? VIEW_DATA : ( window.opener != null && typeof window.opener.VIEW_DATA !== 'undefined' ? window.opener.VIEW_DATA : [] );
} catch (err) {
        console.error(err);
}
```

**Comment by josteink** [comment]
Thank you for very detailed and well documented bug-report. This is way above average and thus I'm exceptionally sorry for the extreme delay in responding to it. I really am. Please accept my deepest apologies for this delay.

I've tried reproducing it, but I honestly can't find a way to trigger email-sending from the contacts-page. I've managed to activate the Contacts page and trigger an email from the *chat* section when trying to reach offline contacts, but that's my best.

From what I can tell, that does not fit your description. Maybe Google changed something, and this is no longer a possible way to send email (and thus crash Gmail.js)?

Basically, at the moment, I'm not able to reproduce, so I'm closing this issue.

Please re-open if you can fully document a way to still reproduce this. Let me assure you,  I honestly appreciate the effort you put into filing this request, and I'm sorry we took so long to respond.

**Comment by tomxhu** [comment]
This issue is related to when the `window.opener` exists and is a cross-origin frame issue.

You can duplicate by installing the widget from above and opening Gmail into a full screen view via a redirect. This JSFiddle should trigger it: https://jsfiddle.net/cqtjeaht/127/

The try catch workaround @avzabr mentioned is a workaround for at least not crashing the script

**Comment by josteink** [comment]
Thank you. That's much more reproducible!

Looking at the workaround though... I'm a little weary. If we can't get `globals` 99% of the remaining code will fail because that is a critical variable we need to process pretty much everything else.

What do we gain by silently "tolerating" this critical error, only to fail down the line?

**Comment by davojta** [comment]
the issue is reproducible and workaround (or view) to wrap just assignment for VIEW_DATA in try / catch.
Should i open new issue for that, @josteink ?

**Comment by josteink** [comment]
Maybe I misunderstand something, but if you ignore that error... GmailJS will have lots of issues *down the line* because it depends on these variables, right?

Or are you claiming that ignoring failure here still keeps GmailJS working?

**Comment by davojta** [comment]
Yes i'm claiming that without view data gmail js is still working in case of an user will open gmail from internal portal for example.

And as you can see in the code 

```
 api.tracker.view_data = typeof VIEW_DATA !== "undefined"
            ? VIEW_DATA
            : (
                typeof(window) !== "undefined" && window.opener !== null && typeof window.opener.VIEW_DATA !== "undefined"
                    ? window.opener.VIEW_DATA
                    : []
```
we assign default zero array if we cant find view_data, but if the user will open gmail js from external site - the gmail js will not inited and will not work. I'm proposing to not throw error in case gmail opened from other website.

**Comment by josteink** [comment]
OK. I'll take your word for it.

As an absolute worst case, I cannot see this change negatively impacting or breaking any code which works now, so if you say this change will help a use-case you have, I guess we have nothing to lose by trying :)

Could you provide your changes as PR?

**Comment by davojta** [comment]
I will try in next days!
